modules/server/src/network/index.ts

```diff
@@ -1,10 +1,73 @@
-import { NetworkAggregationInterface } from '@/config/types';
+import { buildClientSchema, getIntrospectionQuery, IntrospectionQuery } from 'graphql';
+import { fetchGql } from './gql';
+import { NetworkAggregationConfig, NetworkAggregationConfigInput } from './types';
+import { toJSON } from './util';
 
-export const createSchemaFromNetworkConfig = ({
+type FetchRemoteSchemaResult = {
+	config: NetworkAggregationConfigInput;
+	schemaJSON: IntrospectionQuery | null;
+};
+const fetchRemoteSchema = async (
+	config: NetworkAggregationConfigInput,
+): Promise<FetchRemoteSchemaResult> => {
+	const { graphqlUrl } = config;
+	try {
+		/**
+		 * get full schema (needed for buildClientSchema) and convert json
+		 */
+		const remoteSchema = await fetchGql({
+			url: graphqlUrl,
+			gqlRequest: { query: getIntrospectionQuery() },
+		})
+			.then(toJSON)
+			.then((json) => json.data);
+
+		return { config, schemaJSON: remoteSchema };
+	} catch (error) {
+		console.log(`Failed to retrieve schema from url: ${config.graphqlUrl}`);
+		console.error(error);
+		return { config, schemaJSON: null };
+	}
+};
```

Comment:
return `schema: null` on error and log errors
need all queries to run even if errors

Reply:
This kind of information goes nicely into a TSDoc comment before the funciton. Let users know what happens on successful and unsuccessful fetches.